src/notebooks/gen_yolo_training_data.ipynb

```diff
@@ -0,0 +1,234 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "source": [
+    "# An example notebook how to generate training data for YOLOv5 from an .EMObs file"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "outputs": [],
+   "source": [
+    "import os, sys\n",
+    "sys.path.append(\"..\")\n",
+    "\n",
+    "import emtmlibpy.emtmlibpy as emtm\n",
+    "from emtmlibpy.emtmlibpy import EMTMResult\n",
+    "\n",
+    "import pandas as pd\n"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "outputs": [],
+   "source": [
+    "SRC_DIR = os.path.abspath(os.getcwd())\n",
+    "TEST_FILES_PATH = os.path.join(SRC_DIR, '..', 'test_files')"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Sanity check that we can read the library"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(2, 0)\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(emtm.emtm_version())"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Load the .EMObs files"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "outputs": [],
+   "source": [
+    "r = emtm.em_load_data(os.path.join(TEST_FILES_PATH, 'Test.EMObs'))\n",
+    "assert EMTMResult(r) == EMTMResult(0)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Load the data in Test.EMObs into a dataframe"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[(b'', b'', b''), (b'balistidae', b'abalistes', b'stellatus'), (b'nemipteridae', b'nemipterus', b'furcosus'), (b'nemipteridae', b'pentapodus', b'porosus'), (b'pinguipedidae', b'parapercis', b'xanthozona'), (b'scombridae', b'scomberomorus', b'queenslandicus')]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# this needs to be called first to generate the list\n",
+    "n_fgs = emtm.em_unique_fgs()\n",
+    "fgs = []\n",
+    "\n",
+    "for ii in range(n_fgs):\n",
+    "    fgs.append(emtm.em_get_unique_fgs(ii))\n",
```

Comment:
Obviously a matter of taste, but I'd be tempted to use list or set comprehension here. E.g.:
```suggestion
    "fgs = {emtm.em_get_unique_fgs(ii) for ii in range(n_fgs)}\n",
```
(Assuming a set is appropriate for your use.)

Reply:
It's like this right now so I can brearkpoint the list and inspect all the rows